xdsl/transforms/canonicalization_patterns/riscv.py

```diff
@@ -21,6 +21,17 @@ def match_and_rewrite(self, op: riscv.MVOp, rewriter: PatternRewriter) -> None:
             rewriter.replace_matched_op([], [op.rs])
 
 
+class ImmediateMoveToCopy(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv.MVOp, rewriter: PatternRewriter, /):
+        if (
+            isinstance(op.rd.type, riscv.RISCVRegisterType)
+            and not op.rd.type.is_allocated
```

Comment:
I think this should work fine whether or not the result is allocated, as long as the move is to the same register